web/api/v1/api.go

```diff
@@ -492,14 +492,42 @@ func (api *API) labelNames(r *http.Request) apiFuncResult {
 	if err != nil {
 		return apiFuncResult{nil, &apiError{errorBadData, errors.Wrap(err, "invalid parameter 'end'")}, nil, nil}
 	}
+	var matcherSets [][]*labels.Matcher
+	for _, s := range r.Form["match[]"] {
```

Comment:
We now need to complete the discussion that was cut short in #8288, as this should have the same handling of empty selectors as all the other places.